<?xml version="1.0" ?>
<launch>
  <arg name="robot_name"	default="b_bot"/>
  <arg name="camera_name"	default="a_phoxi_m_camera"/>
  <arg name="config"		default="sim"/>
  <arg name="marker_id"		default="26"/>
  <arg name="marker_size"	default="0.070"/>    <!-- in m -->

  <include file="$(find o2as_easy_handeye)/launch/o2as_publish.launch">
    <arg name="robot_name"	value="$(arg robot_name)" />
    <arg name="camera_name"	value="$(arg camera_name)" />
  </include>

  <!-- Set AR marker ID -->
  <!-- Start ArUco -->
  <!-- <node name="aruco_tracker" pkg="aruco_ros" type="single"> -->
  <node name="aruco_tracker" pkg="o2as_aruco_ros" type="o2as_single"
  	output="screen">
    <remap from="/camera_info"	to="/$(arg camera_name)/camera_info"
	   if="$(eval arg('camera_name') == 'a_phoxi_m_camera')"/>
    <remap from="/image"	to="/$(arg camera_name)/texture"
	   if="$(eval arg('camera_name') == 'a_phoxi_m_camera')"/>
    <remap from="/pointcloud"	to="/$(arg camera_name)/pointcloud"
    	   if="$(eval arg('camera_name') == 'a_phoxi_m_camera')"/>
    <remap from="/camera_info"	to="/$(arg camera_name)/rgb/camera_info"
	   unless="$(eval arg('camera_name') == 'a_phoxi_m_camera')"/>
    <remap from="/image"	to="/$(arg camera_name)/rgb/image_raw"
	   unless="$(eval arg('camera_name') == 'a_phoxi_m_camera')"/>
    <remap from="/pointcloud"	to="/$(arg camera_name)/depth/points"
    	   unless="$(eval arg('camera_name') == 'a_phoxi_m_camera')"/>

    <!-- Other parameters -->
    <param name="image_is_rectified" value="false"/>
    <param name="marker_size"        value="$(arg marker_size)"/>
    <param name="marker_id"          value="$(arg marker_id)"/>
    <param name="reference_frame"    value="workspace_center"/>
    <param name="camera_frame"
	   value="camera_link_from_$(arg robot_name)_estimated"/>
    <param name="marker_frame"       value="camera_marker"/>
  </node>

  <!-- Start robot controller -->
  <include file="$(find o2as_examples)/launch/connect_real_robots.launch"
	   if="$(eval arg('config') == 'real')">
  </include>
  <include file="$(find aist_examples)/launch/connect_aist_robots.launch"
	   if="$(eval arg('config') == 'aist')">
  </include>
  <include file="$(find o2as_examples)/launch/connect_osaka_robots.launch"
	   if="$(eval arg('config') == 'osaka')">
  </include>

  <include file="$(find o2as_moveit_config)/launch/o2as_moveit_planning_execution.launch"
  unless="$(eval arg('config') == 'aist')">
    <arg name="sim" value="true"  if    ="$(eval arg('config') == 'sim')"/>
    <arg name="sim" value="false" unless="$(eval arg('config') == 'sim')"/>
  </include>
  <include file="$(find o2as_skills)/launch/o2as_skill_server.launch"
    unless="$(eval arg('config') == 'aist')"/>

  <!-- Start rqt_reconfigure -->
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
</launch>
